src/main/java/org/elasticsearch/node/internal/InternalSettingsPreparer.java

```diff
@@ -38,10 +40,35 @@
  */
 public class InternalSettingsPreparer {
 
+    public static final String SECRET_PROMPT_VALUE = "__prompt:secret__";
```

Comment:
I'm good with that, and a strict YAML parser doesn't barf with this syntax, which is a bonus :)